frontend/src/Components/FavoriteCard.tsx

```diff
@@ -0,0 +1,71 @@
+import { CombinedFavorite } from '@/common';
+import { StarIcon as SolidStar } from '@heroicons/react/24/solid';
+import { useNavigate } from 'react-router-dom';
+
+interface FavoriteCardProps {
+    favorite: CombinedFavorite;
+    onUnfavorite: () => void;
+}
+
+const FavoriteCard: React.FC<FavoriteCardProps> = ({
+    favorite,
+    onUnfavorite
+}) => {
+    const navigate = useNavigate();
+
+    const handleCardClick = () => {
+        if (favorite.visibility_status) {
+            return;
+        }
+        const state = {
+            subLink: favorite.content_url
+        };
+        if (favorite.type === 'video') {
+            navigate(`/viewer/videos/${favorite.content_id}`);
+        } else if (favorite.type === 'library') {
+            navigate(`/viewer/libraries/${favorite.content_id}`, { state });
+        }
+    };
+
+    return (
+        <div
+            className={`card relative rounded-lg p-3 shadow-md ${
+                favorite.visibility_status ? 'bg-grey-2' : 'bg-inner-background'
+            } ${
+                favorite.visibility_status
+                    ? 'cursor-not-allowed'
+                    : 'cursor-pointer '
+            }`}
+            style={{ width: '220px', height: '220px' }}
```

Comment:
also this could be added into the `className` above as `w-[220px] h-[220px]` rather than using the `style` since we don't typically use style.